src/bilder/images/apisix/deploy.py

```diff
@@ -0,0 +1,227 @@
+    dest=f"/opt/cloud-cli-linux-amd64-{VERSIONS['apisix-cloud-cli']}.gz",
+    mode="0644",
+)
+
+server.shell(
+    name="Install apisix cloud-cli binary",
+    commands=[
+        f"/usr/bin/gunzip -c /opt/cloud-cli-linux-amd64-{VERSIONS['apisix-cloud-cli']}.gz > /usr/local/bin/cloud-cli",
+        "chmod a+x /usr/local/bin/cloud-cli",
+    ],
+)
+
+files.put(
+    name="Place the cloud-cli-configure service definition.",
+    src=str(FILES_DIRECTORY.joinpath("cloud-cli-configure.service")),
+    dest="/usr/lib/systemd/system/cloud-cli-configure.service",
+    mode="644",
+)
+files.put(
+    name="Place the cloud-cli-deploy service definition.",
+    src=str(FILES_DIRECTORY.joinpath("cloud-cli-deploy.service")),
+    dest="/usr/lib/systemd/system/cloud-cli-deploy.service",
+    mode="644",
+)
+files.put(
+    name="Place supplemental configuration file.",
+    src=str(FILES_DIRECTORY.joinpath("supplemental_config.yaml")),
+    dest="/etc/docker/config.yaml",
+    mode="644",
+)
+
+
+vault_template_permissions(vault_config)
+
+# Install vector
+install_vector(vector_config)
+configure_vector(vector_config)
+
+# Lay down final configuration for hashicorp products
+for product in hashicorp_products:
+    configure_hashicorp_product(product)
+
+# Place the tags document
+tags_json = json.dumps(
+    build_tags_document(
+        source_tags={
+            "consul_version": VERSIONS["consul"],
+            "vault_version": VERSIONS["vault"],
+            "traefik_version": VERSIONS["traefik"],
+            "apisix_version": VERSIONS["apisix"],
+            "apisix_cloud_cli_version": VERSIONS["apisix-cloud-cli"],
+        }
+    )
+)
+files.put(
+    name="Place the tags document at /etc/ami_tags.json",
+    src=StringIO(tags_json),
+    dest="/etc/ami_tags.json",
+    mode="0644",
+    user="root",
+)
+
+# Setup systemd daemons for everything
+if host.get_fact(HasSystemd):
+    vector_service(vector_config)
+
+    register_services(hashicorp_products, start_services_immediately=False)
+    proxy_consul_dns()
+
+    server.service(
+        name="Ensure docker compose service is disabled",
+        service="docker-compose",
+        enabled=False,  # We won't actually use docker-compose to run this service
+        running=False,
+    )
+
+    server.service(
+        name="Ensure the cloud-cli-configure service is enabled",
+        service="cloud-cli-configure",
+        enabled=True,
+        running=False,
+    )
+    server.service(
+        name="Ensure the cloud-cli-deploy service is enabled",
+        service="cloud-cli-deploy",
+        enabled=True,
+        running=False,
+    )
+
+    watched_docker_compose_files = [
+        DOCKER_COMPOSE_DIRECTORY.joinpath(".env"),
+    ]
+    service_configuration_watches(
+        service_name="docker-compose", watched_files=watched_docker_compose_files
+    )
```

Comment:
Do we actually want to register file watches for bouncing docker-compose since we aren't enabling the service?

Reply:
Good point.